packages/components/src/tools-panel/tools-panel/hook.ts

```diff
@@ -81,44 +97,55 @@ export function useToolsPanel( props ) {
 	};
 
 	// Manage and share display state of menu items representing child controls.
-	const [ menuItems, setMenuItems ] = useState( {} );
+	const [ menuItems, setMenuItems ] = useState< ToolsPanelMenuItems >( {
+		default: {},
+		optional: {},
+	} );
 
 	const getResetAllFilters = () => {
-		const filters = [];
+		const filters: Array< Function > = [];
 
-		panelItems.forEach( ( item ) => {
+		panelItems.forEach( ( item: ToolsPanelItem ) => {
 			if ( item.resetAllFilter ) {
 				filters.push( item.resetAllFilter );
 			}
 		} );
-
 		return filters;
 	};
 
 	// Setup menuItems state as panel items register themselves.
 	useEffect( () => {
-		const items = generateMenuItems( { panelItems, reset: false } );
+		const items: ToolsPanelMenuItems = generateMenuItems( {
+			panelItems,
+			reset: false,
+		} );
 		setMenuItems( items );
 	}, [ panelItems ] );
 
 	// Toggle the checked state of a menu item which is then used to determine
 	// display of the item within the panel.
-	const toggleItem = ( label ) => {
-		const currentItem = panelItems.find( ( item ) => item.label === label );
+	const toggleItem = ( label: string ) => {
+		const currentItem: ToolsPanelItem | undefined = panelItems.find(
+			( item: ToolsPanelItem ) => item.label === label
+		);
```

Comment:
The typings here should not necessary because TypeScript knows that `panelItems` is of type `ToolsPanelItem[]`, and that `Array.find()` can return such an item or undefined 

```suggestion
		const currentItem = panelItems.find( ( item ) => item.label === label );
```